hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/FormatTRPRequestMessage.java

```diff
@@ -0,0 +1,95 @@
+/**
+ *
+ */
+package ca.bc.gov.hlth.hnsecure.parsing;
+
+import static ca.bc.gov.hlth.hnsecure.message.ErrorMessage.HL7_ERROR_NO_INPUT;
+
+import org.apache.camel.Handler;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.bc.gov.hlth.hncommon.util.LoggingUtil;
+import ca.bc.gov.hlth.hnsecure.exception.CustomHNSException;
+
+/**
+ * Purpose of this bean is to convert TRP request message to formal format. The
+ * PNET endpoint expects optional field separators to be appended for MSH, ZZZ
+ * and ZCC segment
+ * https://dev.azure.com/cgi-vic-hlth/HNI%20Modernization/_workitems/edit/10462/
+ *
+ */
+public class FormatTRPRequestMessage {
+
+	private static Logger logger = LoggerFactory.getLogger(FormatTRPRequestMessage.class);
+
+	@Handler
+	public String format(String v2Message) throws CustomHNSException {
+		logger.debug("{}: Converting TRP request to formal format {}", LoggingUtil.getMethodName(), v2Message);
+
+		// It should be impossible for the body to be empty here (the handshake server
+		// should catch that) but handle it just in case
+		if (StringUtils.isBlank(v2Message)) {
+			throw new CustomHNSException(HL7_ERROR_NO_INPUT);
+		}
+
+		String[] segments = V2MessageUtil.getMessageSegments(v2Message);
+		String zzzSegment = V2MessageUtil.getSegment(segments, V2MessageUtil.SegmentType.ZZZ);
+		String[] zzzSegmentFields = zzzSegment.split(Util.DOUBLE_BACKSLASH + Util.HL7_DELIMITER, -1);
+
+		// Apply formal formatting only for TRP transaction id
+		if (zzzSegmentFields.length >= 1 && zzzSegmentFields[1].equalsIgnoreCase("TRP")) {
+
+			// Append missing optional fields
+			if (zzzSegmentFields.length >= 6 && zzzSegmentFields.length < 11) {
+				while (zzzSegmentFields.length < 11) {
```

Comment:
It seems overcomplex to have recursive of length each time when this could be calculated initially and then a string built to replace segment once at the end, preferably when returning the message so that value being passed to the method is not changed.